mpr-research-data.py

```diff
@@ -144,51 +107,228 @@ def sliceAndPushToGCP(courseDF, retrieveDF, client, targetBucketName=targetBucke
+                    allKeyPartsFound = False
+
+            else:
+                try:
+                    self.dbParams[credPart] = str(os.getenv(
+                        'DB_' + credPart, dbCredsDefaultDict[credPart]))
+                except TypeError as e:
+                    logging.error(f'Error Message: {e}')
+                    logging.error(
+                        f'Invalid type (expected str) passed for DB_{credPart}.')
+                    allKeyPartsFound = False
+
+            if not self.dbParams.get(credPart, False):
+                logging.error(
+                    f'Did not find configuration parameter for M-Write Peer Review production DB key: {credPart}.')
+                allKeyPartsFound = False
+
+        return allKeyPartsFound
+
+    def setAndVerifyGCPCreds(self):
+        try:
+            self.gcpParams = json.loads(str(os.getenv('GCP_KEY')))
+            return True
+        except json.JSONDecodeError as e:
+            logging.error(f'Error Message: {e}')
+            logging.error(
+                f'Invalid JSON format passed for GCloud Service JSON Key.')
+            return False
 
-# RETRIEVE COURSE INFO
-# --------------------------------------------------------------------------
-courseQueryDF = courseQueryMaker(
-    queryTemplateDict['course'], numberOfMonths, sqlEngine)
-# sys.stdout.flush()
+    def setFromEnv(self):
 
-# RETRIEVE COURSE DATA
-# --------------------------------------------------------------------------
-retrieveQueryDF = retrieveQueryMaker(
-    queryTemplateDict['retrieve'], courseQueryDF['id'], sqlEngine)
-# sys.stdout.flush()
+        try:
+            self.logLevel = str(os.environ.get(
+                'LOG_LEVEL', self.logLevel)).upper()
+        except TypeError as e:
+            logging.warning(f'Error Message: {e}')
+            logging.warning(
+                'Incorrect type for configuration parameter for log level provided. Defaulting to INFO level.')
+        try:
+            logging.getLogger().setLevel(logging.getLevelName(self.logLevel))
+        except ValueError as e:
+            logging.warning(f'Error Message: {e}')
+            logging.warning(
+                'Invalid configuration parameter for log level provided. Defaulting to INFO level.')
+
+        envImportSuccess = True
+        try:
+            self.targetBucketName = str(
+                os.getenv('GCLOUD_BUCKET', self.targetBucketName))
+        except TypeError as e:
+            logging.error(f'Error Message: {e}')
+            logging.error(
+                f'Invalid type (expected str) passed for GCloud bucket name.')
+            envImportSuccess = False
+
+        try:
+            self.numberOfMonths = int(
+                os.getenv('NUMBER_OF_MONTHS', self.numberOfMonths))
+
+            if self.numberOfMonths < 1:
+                logging.error(
+                    f'Config parameter NUMBER_OF_MONTHS must be >= 1.')
+                envImportSuccess = False
```

Comment:
This check is very good.  I had noticed it myself shortly before I saw you added it.

However, I just tested this check by setting months to 0 and I got this output…

```
mpr-research-data | Starting mpr-research-data.py...
mpr-research-data | 2022-05-13T19:27:46+00:00
mpr-research-data | 2022-05-13T19:27:48+0000 ERROR [mpr-research-data.py:250] - Config parameter NUMBER_OF_MONTHS must be >= 1.
mpr-research-data | 2022-05-13T19:27:48+0000 INFO [mpr-research-data.py:156] - Loading in SQL file - courseQuery.sql
mpr-research-data | 2022-05-13T19:27:48+0000 INFO [mpr-research-data.py:156] - Loading in SQL file - retrieveQuery.sql
mpr-research-data | Exiting due to configuration parameter import problems.
mpr-research-data | 🏁 - Python Script Execution Complete.
mpr-research-data | 2022-05-13T19:27:48+00:00
mpr-research-data exited with code 0
```

It seems to me that the application should've exited after the error about the months and before the queries are loaded.

Reply:
This is partly intentional. All config variables are checked even if one has failed. This way, if multiple config values have issues, they can all be logged, rather than doing it over multiple runs. I can change that if that is not preferred.

Reply:
Oh, I see.  Yes, that's a pretty good approach.  You don't need to change in that case.